src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,41 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents the Contact data type in the address book. 
+ * Is superclass of any user details. 
+ */
+public class Contact {
+    protected String value;
+    protected boolean isPrivate;
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
```

Comment:
Could you have extracted more from the subclasses? Maybe extract the constructors?